src/main/java/com/redhat/devtools/intellij/tektoncd/actions/component/ShowLogsAction.java

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc.
+ ******************************************************************************/
+package com.redhat.devtools.intellij.tektoncd.actions.component;
+
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.redhat.devtools.intellij.common.tree.LazyMutableTreeNode;
+import com.redhat.devtools.intellij.tektoncd.actions.TektonAction;
+import com.redhat.devtools.intellij.tektoncd.tkn.PipelineRun;
+import com.redhat.devtools.intellij.tektoncd.tkn.TaskRun;
+import com.redhat.devtools.intellij.tektoncd.tkn.Tkn;
+import com.redhat.devtools.intellij.tektoncd.tree.PipelineNode;
+import com.redhat.devtools.intellij.tektoncd.tree.PipelineRunNode;
+import com.redhat.devtools.intellij.tektoncd.tree.TaskNode;
+import com.redhat.devtools.intellij.tektoncd.tree.TaskRunNode;
+import com.redhat.devtools.intellij.tektoncd.ui.component.RunPickerDialog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.tree.TreePath;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.redhat.devtools.intellij.tektoncd.Constants.NOTIFICATION_ID;
+
+public class ShowLogsAction extends TektonAction {
+    Logger logger = LoggerFactory.getLogger(ShowLogsAction.class);
+
+    public ShowLogsAction() { super(PipelineRunNode.class, TaskRunNode.class, TaskNode.class, PipelineNode.class); }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Tkn tkncli) {
+        String namespace;
+        String resourceName = selected.toString();
+        List<String> resourceRunsName = null;
+        Class<?> nodeClass = selected.getClass();
+        String kindLabel = "";
+        Notification notification;
+        // if node selected is a Pipeline or a Task we need to find its runs
+        try {
+            namespace = ((LazyMutableTreeNode)selected).getParent().getParent().toString();
```

Comment:
This first part should not be run in the UI thread as it will block IJ if the process is long to complete